### DISCUSSION TITLE
How to create comp.Intel or comp.Gnu file?

### DISCUSSION THREAD

**Discussion by Cesar-IIUNAM** [NOAA-EMC/WW3#1117] [Q&A]
Hello, dear. 
I have been able to compile the latest version of WW3 on my institute's cluster, and I have already done several simulations. 
Finally, my goal is to couple WW3 with CROCO using OASIS.mct. 
However, when I use a provided script to compile WW3 within this coupling architecture and choosing Intel as compiler, I encounter the error that it does not find "comp.Intel".
I have checked /WW3/model/bin and indeed it is not there. 

I compiled the model according to:
"cd WW3"
"mkdir build && cd build"
"cmake ... -DSWITCH=/path/to/switch_NCEP_st2 -DCMAKE_INSTALL_PREFIX=install"
This section, on a pc with ifort as compiler, would look like this:
"cmake ... -DSWITCH= /opt/Data2/ctorresm/WW3/model/bin/ switch_IIUNAM_CLUSTER -DCMAKE_INSTALL_PREFIX=install"
"make"
"make install"

With this, in /WW3/model/bin I have "comp.tmpl" and "link.tmpl".
How can I create the comp.COMPILER file, e.g. for Intel comp.Intel?

Thank you for your attention. 
César.

**Comment by MatthewMasarik-NOAA** [comment]
Hi @Cesar-IIUNAM, could you share the error output your seeing from the calls above?  I don't think you should have to worry about those files (comp.tmpl, link.tmpl) using the cmake build.

**Reply by Cesar-IIUNAM** [reply]
Hi @MatthewMasarik-NOAA, thank you for your answer. 
I mean, when I try to compile WW3 for coupling with CROCO via OASIS, the coupling code looks for a "comp.COMPILER" file, e.g. for Intel "comp.Intel" or gfortran "comp.Gnu", but when I compiled WW3 initially (as I show above), none of these files were generated. So I wonder, is it possible to generate them even after compiling the model initially?

**Reply by MatthewMasarik-NOAA** [reply]
Hi @Cesar-IIUNAM, okay thank you for clarifying.  I admittedly have limited knowledge of OASIS, and even less of CROCO, so I'm probably not the best equipped to troubleshoot here.

Hopefully others in the community that have experience with OASIS and CROCO can chime in?

**Reply by Cesar-IIUNAM** [reply]
@MatthewMasarik-NOAA, don't worry, I thank you for your support.
In fact, I ask the question in this forum, because for previous versions of WW3 I could see these files that I mention. For instance, for version 6.07 I have some compilation instructions as the following: 

git clone https://github.com/NOAA-EMC/WW3/archive/refs/tags/6.07.1.tar.gz
export WWATCH3_NETCDF=NC4
export NETCDF_CONFIG=/usr/bin/nc-config
cd ./WWW3-6.07.1/model/bin/
cp comp.Gnu comp
cp link.Gnu link
cp switch_Ifremer2 switch 
./w3_setup ../
./w3_make

And with this it was possible to have comp.Gnu and link.Gnu
Any guidance from other friends in the community would be greatly appreciated.

Regards,

**Comment by mickaelaccensi** [comment]
Dear Cesar,

you should use cmake to compile WW3. Please follow the regtest ww3_tp2.14 which use oasis mct with cmake

**Comment by Cesar-IIUNAM** [comment]
@mickaelaccensi thank you very much for your reply. 
In fact I have tried to compile with cmake. I am doing the following:

I created the OASISDIR environment variable in my bashrc like this:
export OASISDIR=/opt/Data2/ctorresm/oasis/compile_oa3-mct

Then:
cd WW3
mkdir build && cd build (I already had this directory created)
module load cmake 
cmake .. -DSWITCH=/opt/Data2/ctorresm/WW3/model/bin/switch_IIUNAM_OW -DCMAKE_INSTALL_PREFIX=install
make

It looks like WW3 knows where to find the OASIS directory, as indicated in the screenshot below, but when I do "make" it processes for a while and then I get some errors, which I summarize in the attached document 
[error_compiling_WW3-OASIS-mct.txt](https://github.com/NOAA-EMC/WW3/files/13284417/error_compiling_WW3-OASIS-mct.txt).

<img width="1111" alt="image" src="https://github.com/NOAA-EMC/WW3/assets/113042121/4e7bfec0-c1a3-4057-832a-e2bcd04103e1">

Thanks for any suggestions you can give me.

**Comment by fakaaaaaaaa12** [comment]
Luckily we had a similar problem and unfortunately I'm still trying to work out a solution as well:
My progress and intentions so far: I downloaded OASIS3-MCT5.0 (the latest version so far) from the official website and finished compiling it (via the mpi built-in in the OASIS3-MCT5.0 installer, i.e. openmpi 4.0.3).
Currently trying to compile WW3 with the same compiler (via switch_OASACM), since all previous use of WW3 was compiled via Gnu, the error you reported should also be due to unrecognized related libraries due to a different compiler (or version), I'm getting a very similar error message as you, and to solve the problem, I think I need to first determine the environment variable for OASISDIR, the At the same time, the comp and link files should be set to correspond to the selected settings, I can only think of so much at the moment, if you also have the latest progress, I think we can actively exchange!
[error_compiling_WW3-OASIS-mct.txt](https://github.com/NOAA-EMC/WW3/files/13759206/error_compiling_WW3-OASIS-mct.txt)

**Reply by Cesar-IIUNAM** [reply]
hi @fakaaaaaaaa12. Thank very much for your message and apologies for the delay in responding; I was on a long vacation, but I'm back to work. 
I don't know if you were able to solve the problem. For my part, I'm still having the same error. In the first instance, I have already compiled OASIS-MCT and I have my libraries and directories defined like this:

# MPI library
#MPI = $(shell which ifort)
MPIDIR = /opt/intel/oneapi/mpi/2021.9.0
MPIBIN = $(MPIDIR)/bin
MPI_INCLUDE = $(MPIDIR)/include
MPILIB = -L$(MPIDIR)/lib
MPIRUN = $(MPIBIN)/mpirun
#
# Compilers and options
MAKE = gmake
F90 = $(MPIBIN)/mpiifort -I$(MPI_INCLUDE)
F = $(F90)
f90 = $(F90)
f = $(F90)
CC = $(MPIBIN)/mpiicc -I$(MPI_INCLUDE)
LD = $(MPIBIN)/mpiifort $(MPILIB)

Next, I am trying to compile WW3 with switch_OASOCM, but I get the error you have already seen. When compiling using cmake, I must admit that I don't know which compiler is being used. 

Any comments would be greatly appreciated.
Cesar.
